lib/ValgrindBenchmarkTools/src/ValgrindBenchmarkTools.jl

```diff
@@ -0,0 +1,30 @@
+module ValgrindBenchmarkTools
+
+using Valgrind, Reexport
+@reexport using BenchmarkTools
+
+@inline function prehook()
+    Valgrind.Callgrind.zero_stats()
+    Valgrind.Callgrind.start_instrumentation()
+end
+
+@inline function posthook(id)
+    Valgrind.Callgrind.stop_instrumentation()
+    Valgrind.Callgrind.dump_stats_at(id)
+end
+
+function __init__()
+    BenchmarkTools.DEFAULT_PARAMETERS = BenchmarkTools.Parameters(;
+        prehook = prehook,
+        posthook = posthook,
+        enable_customizable_func = :ALL,
+    )
```

Comment:
If you don't want to run the normal benchmarking BenchmarkTools does you can set `run_customizable_func_only` to false.